compiler/rustc_macros/src/session_diagnostic.rs

```diff
@@ -105,10 +105,16 @@ impl SessionDiagnosticDeriveError {
     }
 }
 
+#[cfg(not(bootstrap))]
 fn span_err(span: impl proc_macro::MultiSpan, msg: &str) -> proc_macro::Diagnostic {
     Diagnostic::spanned(span, proc_macro::Level::Error, msg)
 }
 
+#[cfg(bootstrap)]
+fn span_err(span: impl proc_macro::Spanned, msg: &str) -> proc_macro::Diagnostic {
+    Diagnostic::error(msg).mark_all(span)
+}
```

Comment:
```suggestion
#[cfg(bootstrap)]
fn span_err(span: impl proc_macro::MultiSpan, msg: &str) -> proc_macro::Diagnostic {
    Diagnostic::spanned(span, proc_macro::Level::Error, msg)
}

#[cfg(not(bootstrap))]
fn span_err(span: impl proc_macro::Spanned, msg: &str) -> proc_macro::Diagnostic {
    Diagnostic::error(msg).mark_all(span)
}
```

Weird, seemed reversed to me, the errors seemed to show this part as well. Can you please try this?

Reply:
That's what I tried at first. It didn't succeed locally.

Reply:
Not sure, maybe need someone else to help. cc @jyn514 to see if he can help.

Reply:
`rustc_macros` is a proc macro crate, cargo may pass rustc flags differently to it, so I'm not even sure whether `bootstrap` is passed to it at all.
If you don't want to debug the build issues, keeping the old API around will work around this problem for sure.

Reply:
Can't keep the entirety of the old API around, as some methods (such as `Diagnostic::error`) have the same name but change signatures. I know this because I've already tried :laughing:

Reply:
I opened https://github.com/rust-lang/rust/pull/83433 which should fix this.